tests/Clinic.UnitTests/ServiceTests/OfficeServiceTests.cs

```diff
@@ -1,13 +1,183 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿using AutoMapper;
+using Clinic.API.Mapper;
+using Clinic.BLL.Models;
+using Clinic.BLL.Services;
+using Clinic.DAL.Entities;
+using Clinic.DAL.Interfaces;
+using Clinic.UnitTests.TestData.Offices;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using Shouldly;
+using System;
+using System.Net.Sockets;
 
 namespace Clinic.UnitTests.ServiceTests
 {
     public class OfficeServiceTests
     {
+        private readonly IMapper _mapper;
+        private readonly IOfficeRepository _officeRepository;
+        private readonly OfficeService _officeService;
 
+        public OfficeServiceTests()
+        {
+            var mappingConfig = new MapperConfiguration(config =>
+            {
+                config.AddProfile(new MapperProfile());
+            });
+            IMapper mapper = mappingConfig.CreateMapper();
+            _mapper = mapper;
+
+            _officeRepository = Substitute.For<IOfficeRepository>();
+            _officeService = new OfficeService(_officeRepository, _mapper);
+        }
+
+        [Fact]
+        public async Task GetById_ValidId_ReturnOffice()
+        {
+            //Arrange
+            var id = new Guid();
+            var office = TestOfficeModels.Office;
+
+            _officeRepository.GetByIdAsync(id, default).Returns(office);
+
+            //Act
+            var result = await _officeService.GetByIdAsync(id, default);
+
+            //Assert
+            result.ShouldNotBeNull();
+            result.Address.ShouldBeEquivalentTo(office.Address);
+        }
+
+        [Fact]
+        public async Task GetById_InvalidId_ReturnNull()
+        {
+            //Arrange
+            var id = new Guid();
+            _officeRepository.GetByIdAsync(id, default).ReturnsNull();
+
+            //Act
+            var result = await _officeService.GetByIdAsync(id, default);
+
+            //Assert
+            await _officeRepository.Received(1).GetByIdAsync(id, default);
+            result.ShouldBeNull();
+        }
+
+        [Fact]
+        public async Task GetAllAsync_ByIsActive_ReturnCollectionOfOffices()
+        {
+            //Arrange
+            var office = _mapper.Map<IEnumerable<OfficeEntity>>(TestOfficeModels.SortOffices(null, null, OfficeStatus.Active, default));
+
+            _officeRepository.GetAllAsync(null, null, OfficeStatus.Active, Arg.Any<CancellationToken>()).Returns(office);
+
+            //Act
+            var result = await _officeService.GetAllAsync(null, null, OfficeStatus.Active, new CancellationToken());
+
+            //Assert
+            await _officeRepository.Received(1).GetAllAsync(null, null, OfficeStatus.Active, Arg.Any<CancellationToken>());
+            result.Count().ShouldBeEquivalentTo(1);
+        }
+
+
+        [Fact]
+        public async Task CreateAsync_ReturnCreatedOffice()
```

Comment:
In those cases better to use smth like CreateAsync_ValidData_ReturnCreatedOffice